### DISCUSSION TITLE
2FA

### DISCUSSION THREAD

**Discussion by Rockyread** [community/community#63784] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

The mandatory 2FA requirement of GitHub violates essential Human Rights.
It forces me to buy products I do not wish to buy.
It impairs free speech.
It excludes older people from participating.
There must be other more liberal ways of contributing code to the world.

**Comment by mehulrajdev** [comment, marked as answer]
Two-factor authentication does not impact free speech in any way. It's merely an additional security measure to ensure the rightful owner is accessing the account.

The exclusion of older people is indeed a valid concern. However, it's not so much the 2FA that's excluding, but rather the broader issue of digital literacy among the older population. Efforts can be made to educate and assist them in navigating these security measures.

**Reply by Rockyread** [reply]
ok

**Reply by joeyipjoeyip** [reply]
No reason to not include email 2FA

**Reply by Harem-Rose** [reply]
2FA is only a security measure for people who want it. For others, it is a security risk. There are numerous attack methods employed by bad actors around the world that take advantage of 2FA. 
Right now I need 1 Key to get into my account. A password that is unique to this website and located only in my head. With 2FA I need 2 keys, the aforementioned password, and a physical device that exists in a space where it can be stolen. Not only is it a massive inconvenience, it also increases the threat surface area of my account. 

I already have to verify with an emailed code when I log into the website. That's a 2nd factor already. That should be sufficient. Requiring specialized equipment to access the account is overkill. Additionally, not everyone can have their phone on them at all times. I know I can't. 9 hours a day I cannot have my phone. I may end up in a country long term where my phone doesn't work. In that case, I'm stuck without my account until I'm able to return to my home country. 
Forced 2FA is unacceptably draconian. It's fine to allow it, but it should never be a forced requirement.

**Reply by iSharingan** [reply]
Your required 2FA BS assumes everyone on Github is part of a company and/or produces production code than needs securing (in which case why would those repos be publicly visible in the first place for malicious actors to want to access/modify? No sane person would do that and those accounts would already have 2FA). There are many users who host personal open source projects that are free for anyone to use/modify/submit ideas for changes to and/or isn't they type of code that would be complied into a program.

Requiring 2FA for ALL accounts is an unnecessary hassle, requires buying extra devices in many cases (which the user may or may not have the money, desire, or any other purpose to do so. Why would I [in a hypothetical situation] buy a [new] phone just to use 2FA when I literally do not need the phone for anything else [IE: user owns an older phone that does not support the 2FA app but does everything else they need]), requires an external app that is actually another vulnerability due to lost/stolen devices (or restricted access to the device while on a trip/the device physically will not be present for an extended time), and the app itself never being perfectly secure since the devices it is installed on can never be 100% impervious to malicious actors. Similarly, there are countless old repos with very useful, free-to-use code by users who simply stopped using GitHub years ago and just never logged back in. Your half-baked, braindead policy that can ultimately result in the deletion of that content (directly or indirectly) if the users don't want/don't set up 2FA will kill many projects before they get started when access to those resources is cut off.

Personal account security options like 2FA should remain just that. PERSONAL choices regarding their own OPTIONAL additional account security measures. Do yourself a favor and stop listening to whatever clueless marketing bean-counter is trying to make you Bud-Light yourselves.

**Reply by ghost** [reply]
Why is MicroShaft telling us what we can, or should do with our code

**Reply by ghost** [reply]
**I DON'T WANT 2FA _EVER_!!! PERIOD. FULL STOP.** I **WILL NOT** support **_anything_** requiring my phone number, or another device, just to login and edit **MY PERSONAL PROGRAMS AND PERSONAL FORKS** of programs. I don't need your paranoid worldview.

**Reply by ghost** [reply]
> However, it's not so much the 2FA that's excluding...

Bullshit. 2FA requires, from among other things, your phone number. I don't give my phone number to the fucking physical stores I go to that ask, why the **_hell_** would I give it to faceless online companies that want to track you and remove anonymity? Fuck that shit. This is my **_PERSONAL_** account, and should have the **_full right_** to refuse 2FA for **_ANY REASON AT ALL_ PERIOD**.

By requiring 2FA, you're **_excluding_** anyone who wishes to keep their privacy, and not be tracked.

**Reply by drankinatty** [reply]
The bottom line, 2FA is unnecessary. It does not prove I am me any better than my private key does. It creates impediments to using github. It opens additional software supply-chain vulnerabilities for all users now forced to incorporate another app, or extension with automatic-updates into their computing environment. The "Authenticator" recommended on the 2FA start page uses `npm` of all things to pull in yet more build requirements. Another "Glass Castle" of software built on "Clay Feet". There are always alternatives to github.

**Reply by RealSuperRyn** [reply]
I would be completely fine with this 2FA nonsense if GitHub provided the ability to use an **email account** for 2FA. Requiring that I use a phone implies that you require people who do not use phones that they buy one. "but you can use a physical key" is not an excuse either, some operating systems will not function with them.

**Reply by Anpanator** [reply]
They do not require a phone. You can use TOTP on your PC. If it's the same PC as the one with repo access, however, the security gain is... questionable.

My main gripe is that they don't allow all 2FA methods as primary methods (e.g. Security Keys), so I'll most likely be migrating elsewhere by the end of the month.

**Reply by technik** [reply]
This is not an answer, it's just corporate BS for "yeah, we already made our minds, so we couldn't care less what you say".

**Reply by patrickbnu** [reply]
Or when you want to use your phone number but your country (Brazil) ins't listed to send a sms?

**Reply by wa7jos** [reply]
You can take your 2FA and shove it where the sun don't shine!  I only have ONE project archived and it can sit there forever - I don't care.
The only reason I *HAVE* a cell phone is so I can do 2FA at other sites, which I find cumbersome and intrusive.  Not interested in more of the same.

**Reply by SirFalc0n** [reply]
2FA is gutter trash and so is whatever idiot thought it was a good idea. It should be an option, not a requirement. Down with github.

**Comment by ghost** [comment]
I will not give you my phone number to host my projects, nor will i install an app on my phone. If you're telling me I won't be able to take account actions at a certain date, as in deleting my account, Just delete it now.

intrusive, unnecessary

**Reply by ghost** [reply]
"An additional security measure to ensure the rightful owner is accessing the account. That is being forced down everyone's throat whether they want it or not, Or be restricted from account actions"

I wont be played with, whatever you stand to gain enough to "REQUIRE" it over offering it as an optional security measure

**Reply by Rockyread** [reply]
what ??

**Reply by ghost** [reply]
I just wanted to stop in and say "Fuck you" to any in fact all of the people who thought forcing this shit on everybody was a good idea.
You can burn in hell right along with the fucks that thought "It isn't enough to just allow symbols in passwords, we require them"

**Reply by jmcunx** [reply]
The only option that will work for me is 2FA via email as suggested by RealSuperRyn.  That is because I never give my Cell Phone Number out plus I will not by hardware for the other options.

So, unless email 2FA option is not available, I will stop using github.  I already copied my projects elsewhere to see how that works.  If the email option is not allowed, I will delete my ID from github on Oct 11.

**Comment by omendezmorales** [comment]
Hello,
I setup 2FA twice in my authenticator app, and now github is showing me an error when I use the code provided by _authenticator_: 
![image](https://github.com/community/community/assets/6626759/6b3547e1-31f9-466f-b6f2-ff4e2fbcecc9)
Is there a fix for this ? Thanks,

**Reply by zaher** [reply]
Happened to me too, failed to auth, i used Google Auth, MS Auth and FreeOTP

**Reply by omendezmorales** [reply]
After clicking here and there, I managed to fix it> my Google authenticator needed to be resynchronized, as explained on _[My Google Authenticator codes don’t work](https://support.google.com/accounts/answer/185834?hl=en)_. After that,  the codes generated  worked again :-)

**Reply by twome** [reply]
Wow, it "failed", what an incredibly useful error message. Top-echelon user interface design from Microsoft, as usual.

**Reply by GuillermoFA** [reply]
:D

**Comment by kikfox** [comment]
Github's 2FA is nothing to do with "security" blah blah ... they just want to track you ! They already implemented **Geofencing**  and "**Shadow Banning Users**" .

**Comment by n1kdo** [comment]
![image](https://github.com/community/community/assets/18338660/dd0bf460-2c5c-4e80-9497-9bda452d03d3)


This is completely useless.

**Comment by mirddes** [comment]
fucking useless cunts; EMAIL IS ALREADY 2FA.

FUCKING CUNTS.

**Comment by rsv-code** [comment]
When is it enough? 2, 3, 4 factor, face scan ... Instead of inconveniencing and annoying users by mandating a hack on top of an already broken system maybe github should participate in a better solution.

**Reply by r57zone** [reply]
After scanning the face, they will want to scan the ass, all for safety.

**Comment by ghost** [comment]
> we need id verification so github users will be safe!

WTF.do.you.think. a username and password, e-mail address, SSH key, and **_ALL_** that other shit is?!?! That's like saying that your fingerprint is a password... **NO!!!** it's a user ID, as it's unique to the person, and the password is the verification of that ID.

**Reply by ghost** [reply]
Ah, ok. Can't tell if it's real or sarcasm anymore... it's sad that so many times comments like that aren't just sarcastic irony, but their serious thoughts.

**Comment by r57zone** [comment]
Totally support, whoever wants to let them use 2FA, whoever doesn't want to give the option to not use it.

**Comment by Petschko** [comment]
I will not use it! Bye github, block me i dont care, then i host my stuff somewhere else lol~

**Comment by tradesouthwest** [comment]
Now that Microshaft owns github; it will forever be running commercial logic on all assets. Ads are probably next but I think they are more interested in stealing Open source work that is done by the Open source Community---for profit---rather than keeping it Open source.

This is obviously what is happening. WANTING (requiring) my phone number is just proof of this sham.

**Comment by r57zone** [comment]
Microsoft is destroying another product, a worthless policy of stupid people. They don't care about security, they want to force developers to use this 2FA shit apps.
I wish Microsoft bankruptcy.

**Comment by metux** [comment]
I've written a script to run it all on my local machine:

https://gist.github.com/metux/ce43da58d9759019a0fbf4d25ccb1b00

Since quite recently, the QR image decoding isn't really needed anymore, since github added a link to show the raw key on setup.

**Reply by Iron-E** [reply]
You saved me a lot of time in avoiding having to port over my repos to GitLab. Thank you.

**Comment by howtophil** [comment]
> The bottom line, 2FA is unnecessary. It does not prove I am me any better than my private key does. It creates impediments to using github. It opens additional software supply-chain vulnerabilities for all users now forced to incorporate another app, or extension with automatic-updates into their computing environment. The "Authenticator" recommended on the 2FA start page uses `npm` of all things to pull in yet more build requirements. Another "Glass Castle" of software built on "Clay Feet". There are always alternatives to github.

The admins marked anything negative as "disruptive" because they know 2FA is bullshit

**Comment by csyezheng** [comment]
I made a command line tool to generate and verify one-time passwords. It is no phone required and is easy to use.

Add security key for the first time
```
a2fa add [flags] <account name> <secret key>
```

Every time you need generates an authentication code, just run the following command:
```
a2fa list
```

a2fa’s github repository:
[https://github.com/csyezheng/a2fa](https://github.com/csyezheng/a2fa)

**Comment by ghost** [comment]
My main account just got sent to limbo after activating 2FA
https://github.com/orgs/community/discussions/73451

**Comment by howtophil** [comment]
Exactly.

On Mon, Dec 11, 2023 at 4:13 PM Carmelo Fernandez-Aguera <
***@***.***> wrote:

> This is not an answer, it's just corporate BS for "yeah, we already made
> our minds, so we couldn't care less what you say".
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/63784#discussioncomment-7824385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMVH3KQKSYCT73LGPSQ2WDYI5ZOFAVCNFSM6AAAAAA3Q2CDA2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TQMRUGM4DK>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Phillip J Rhoades
HowToPhil.com
KE8GGD

**Comment by naourass** [comment]
Either support SMS 2FA for all countries, or provide your own Authenticator App / Browser Extension. I'm not going to use a third party for authenticating to Github, and you don't have SMS support for my country. I'm really confused and I may jump ship.
